server-spi-private/src/main/java/org/keycloak/services/resource/AccountResourceSpi.java

```diff
@@ -0,0 +1,34 @@
+package org.keycloak.services.resource;
+
+import org.keycloak.provider.Provider;
+import org.keycloak.provider.ProviderFactory;
+import org.keycloak.provider.Spi;
+
+/**
+ * <p>A {@link Spi} to replace Account resources.
+ *
+ * <p>Implementors can use this {@link Spi} to override the behavior of the Account endpoints and resources by
+ * creating JAX-RS resources that override those served at /account by default.
+ */
+public class AccountResourceSpi implements Spi {
+
+    @Override
+    public boolean isInternal() {
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return "account-console-extension";
```

Comment:
Can we make more explicit in the name that it is related to the backend of the account? Something like `acount-api-extension`

Reply:
Since the implementations can fully replace the frontend and backend of the account using this mechanism, I thought "console" would be more informative. However, I defer to the team for naming preferences, as I'm not too familiar with what has been done before.